### PR TITLE
Swap python linter and formatter to ruff

### PR DESCRIPTION
**Pull request by janbridley**
## Description

`Flake8` and `yapf` have been replaced with a minimal `ruff` config. Autofixes have been applied, and manual fixes have been implemented where necessary. `pre-commit` has been updated to use ruff.

## Motivation and context

In addition to the increased performance, using `ruff` brings fresnel's python tooling up to spec with most other glotzerlab packages. Note that the config options selected here match the previous flake8 selections, with the exception of the RST rules which are not supported. Additional (more opinionated) options can be added in a future PR.

## Change log

```
Switched python linter and formatter to Ruff
```

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/fresnel/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Fresnel Contributor Agreement**](https://github.com/glotzerlab/fresnel/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of authors](https://github.com/glotzerlab/fresnel/blob/master/doc/credits.rst).

### CODE REVIEW

File: ruff.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+include = ["*.py", "*.ipynb"]
+exclude = [".git","__pycache__","doc/conf.py","run-clang-format.py","build"]
+line-length = 80
```
Comment (joaander) [inline ruff.toml]:
```suggestion
line-length = 88
```

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks, this is great!

Let's embrace the black-like default config and set the line length to 88. With the popularity of black, most users likely have python rulers set to 88 these days.

We should also update `style.rst` to reference `ruff` instead of `yapf` and remove references to the yapf plugins.

**Comment by janbridley** [comment]
> Thanks, this is great!
> 
> Let's embrace the black-like default config and set the line length to 88. With the popularity of black, most users likely have python rulers set to 88 these days.
> 
> We should also update `style.rst` to reference `ruff` instead of `yapf` and remove references to the yapf plugins.

Done - everything should be sorted now!

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**